lua/packer/compile.lua

```diff
@@ -41,21 +41,28 @@ end
 ]]
 
 local module_loader = [[
+
+local function escape_pattern(text)
+    return text:gsub("([^%w])", "%%%1")
+end
+
+local lazy_load_called = {}
+local to_load = {}
+
 local function lazy_load_module(module_name)
-  if module_name == 'packer.load' then return nil end
-  local to_load = {}
-  local i = 1
+  if module_name == 'packer.load' or lazy_load_called[module_name] then return nil end
+  lazy_load_called[module_name] = true
   for module_pat, plugin_name in pairs(module_lazy_loads) do
-    if not _G.packer_plugins[plugin_name].loaded and string.match(module_name, "^" .. module_pat) then
-      to_load[i] = plugin_name
+    if not _G.packer_plugins[plugin_name].loaded and string.match(module_name, "^" .. escape_pattern(module_pat)) then
```

Comment:
Could `escape_pattern` be replaced with `vim.pesc`?

Reply:
Yep just tested it and it works the same 👍🏿 hadn't used that one before 😄